crates/uv-resolver/src/requires_python.rs

```diff
@@ -820,4 +833,20 @@ mod tests {
             }
         }
     }
+
+    #[test_case("==3.12", true)]
+    #[test_case("==2.7", true)]
+    #[test_case("==3.12.1", false)]
+    #[test_case("==3.12.*", false)]
+    #[test_case("==3.*", false)]
+    #[test_case(">=3.10", false)]
+    #[test_case(">3.9", false)]
+    #[test_case("<4.0", false)]
+    #[test_case(">=3.10, <3.11", false)]
+    #[test_case("", false)]
```

Comment:
I know we're using this macro in at least one other place too, but I'd skip on this and just write out the code instead.

```rust
#[test]
fn is_matching_without_patch() {
    let test_cases = [
        ("==3.12", true),
        ("==2.7", true),
        ("3.12.1", false),
        // ...
    ];
    for (version, expected) in test_cases {
        let version_specifiers = VersionSpecifiers::from_str(version).unwrap();
        let requires_python = RequiresPython::from_specifiers(&version_specifiers).unwrap();
        assert_eq!(requires_python.is_matching_without_patch(), expected, "version: {version}");
    }
}
```

The main reason for this is that it would be great to reduce our dependence on proc macros in service of helping compile times.